addons/mail/static/src/discuss/call/common/rtc_service.js

```diff
@@ -1305,31 +1314,61 @@ export class Rtc extends Record {
                 Object.assign(this.state, {
                     sourceScreenStream: sourceStream,
                     screenTrack: outputTrack,
+                    screenAudioTrack: screenAudioTrack,
                     sendScreen: Boolean(outputTrack),
                 });
                 break;
             }
         }
+        if (this.state.screenAudioTrack) {
+            this.updateAudioTrack();
+        }
     }
 
-    async resetAudioTrack({ force = false }) {
-        if (this.state.audioTrack) {
-            this.state.audioTrack.stop();
-            this.state.audioTrack = undefined;
+    async updateAudioTrack() {
+        const { micAudioTrack, screenAudioTrack } = this.state;
+        if (!micAudioTrack && !screenAudioTrack) {
+            return;
+        }
+        if (micAudioTrack && screenAudioTrack) {
+            await this.audioContext?.close();
+            this.audioContext = new AudioContext();
+            const micSource = this.audioContext.createMediaStreamSource(
+                new MediaStream([micAudioTrack])
+            );
+            const screenSource = this.audioContext.createMediaStreamSource(
+                new MediaStream([screenAudioTrack])
+            );
+            const destination = this.audioContext.createMediaStreamDestination();
+            micSource.connect(destination);
+            screenSource.connect(destination);
+            this.state.audioTrack = destination.stream.getAudioTracks()[0];
+        } else {
+            this.state.audioTrack = micAudioTrack ?? screenAudioTrack;
         }
+        await this.network?.updateUpload("audio", this.state.audioTrack);
+    }
+
+    async resetAudioTrack({ force = false }) {
+        this.state.micAudioTrack?.stop();
+        this.state.micAudioTrack = undefined;
+        this.state.screenAudioTrack?.stop();
+        this.state.screenAudioTrack = undefined;
```

Comment:
`resetAudioTrack()` is only for the microphone (maybe it should be renamed), if you stop the `screenAudioTrack` here, screen audio will be tied to the lifespan of the microphone audio (which I don't think is expected).

For example, if you change the setting of your audio input device, `resetAudioTrack()` will be called to get a new one from the right device, but there it will also stop the screen audio.